### DISCUSSION TITLE
Size mismatch between Central Header and Local Header

### DISCUSSION THREAD

**Discussion by reachforvenkat** [icsharpcode/SharpZipLib#684] [Q&A]
We are getting a size mismatch error but the size in central header and local header differs by only a few kilobytes.

Opening the same file in 7-Zip and running a test, there is no error

Snippet from Zipfile.cs
```
				// Size can be verified only if it is known in the local header.
				// it will always be known in the central header.
				if (((localFlags & (int)GeneralBitFlags.Descriptor) == 0) ||
					((size > 0 || compressedSize > 0) && entry.Size > 0))
				{
					if ((size != 0)
						&& (size != entry.Size))
					{
						throw new ZipException(
							string.Format("Size mismatch between central header({0}) and local header({1})",
								entry.Size, size));
					}
```
Is this check mandatory ? Can it be turned off or if the zip can be transformed to remove this error, please advise ?

The zip file is created by standard .Net compression library.

**Comment by piksel** [comment]
It's a sanity check, but I agree that it should be possible to ignore it if you are trying to recover information from a corrupt archive. If the correct value is in the local header, you can use `ZipInputStream` to extract it instead of `ZipFile` since it only uses the local header.

**Reply by piksel** [reply]
Would you mind uploading an affected file to https://archivediag.azurewebsites.net/ ? It will generate a file structure report for it. Note that the user interface is horrible right now. It will just say "blob not found" until the report has been created, then it will show a link to the report.
